src/Network/Minio/S3API.hs

```diff
@@ -358,3 +366,31 @@ headBucket bucket = headBucketEx `catches`
                                    , riBucket = Just bucket
                                    }
       return $ NC.responseStatus resp == HT.ok200
+
+makePresignedURL :: UrlExpiry -> HT.Method -> Maybe Bucket -> Maybe Object
+                 -> Region -> HT.Query -> HT.RequestHeaders -> Minio ByteString
+makePresignedURL expiry method bucket object region queryParams headers = do
+  ci <- asks mcConnInfo
+
+  let
+    hostHeader = (hHost, formatBS "{}:{}" [connectHost ci,
+                                           show $ connectPort ci])
+    ri = def { riMethod = method
+             , riBucket = bucket
+             , riObject = object
+             , riQueryParams = queryParams
+             , riHeaders = hostHeader : headers
+             , riRegion = Just region
+             }
+
+  signPairs <- liftIO $ signV4 ci ri (Just expiry)
+
+  let
+    qpToAdd = map (\(x, y) -> (x, Just y)) signPairs
```

Comment:
Functor instance of `(a,)` is the same as above. So we can do `qpToAdd = fmap Just signPairs`